devices.js

```diff
@@ -9235,6 +9235,22 @@ const devices = [
             await endpoint.read('genPowerCfg', ['batteryPercentageRemaining']);
         },
     },
+    {
+        fingerprint: [{modelID: 'RC-EF-3.0', manufacturerName: 'HEIMAN'}],
+        model: 'HM1RC-2-E',
+        vendor: 'HEIMAN',
+        description: 'Smart remote controller',
+        fromZigbee: [fz.battery, fz.legacy.heiman_smart_controller_armmode, fz.command_emergency],
```

Comment:
Replace `fz.legacy.heiman_smart_controller_armmode` with `fz.command_arm`

Reply:
I replaced the `fz.legacy.heiman_smart_controller_armmode` with `fz_command_arm`. I don't understand though why I should change the `e.action` because this device has exactly these 4 buttons and I think it sends these 4 actions.

However, with the replacement, now I get multiple mqtt messages instead of a single one whenever I press a button. This worked perfectly with the `fz.legacy.heiman_smart_controller_armmod`...

Please advise on what to do.

Reply:
Could you provide the debug log of this?

To enable debug logging set in `configuration.yaml`:
```yaml
advanced:
  log_level: debug
```

Reply:
It seems that the 'repeated mqtt messages' problem also occurs with the `fz.legacy.heiman_smart_controller_armmode`. I wanted to be really sure, and tested for both cases. It seems to make no difference. Apparently (as far as I understand the log), the device sends multiple messages when I press a button a single time :-( 

Here is the log (single press of top button of the remote):

```
Zigbee2MQTT:debug 2021-01-31 21:52:45: Received Zigbee message from 'remote', type 'commandArm', cluster 'ssIasAce', data '{"armmode":1,"code":"","zoneid":255}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-01-31 21:52:45: MQTT publish: topic 'zigbee2mqtt/remote', payload '{"action":"arm_day_zones","action_code":"","action_zone":255,"battery":100,"last_seen":"2021-01-31T21:52:45+01:00","linkquality":47}'
Zigbee2MQTT:debug 2021-01-31 21:52:48: Received Zigbee message from 'remote', type 'commandArm', cluster 'ssIasAce', data '{"armmode":1,"code":"","zoneid":255}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-01-31 21:52:48: MQTT publish: topic 'zigbee2mqtt/remote', payload '{"action":"arm_day_zones","action_code":"","action_zone":255,"battery":100,"last_seen":"2021-01-31T21:52:48+01:00","linkquality":47}'
Zigbee2MQTT:debug 2021-01-31 21:52:51: Received Zigbee message from 'remote', type 'commandArm', cluster 'ssIasAce', data '{"armmode":1,"code":"","zoneid":255}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-01-31 21:52:51: MQTT publish: topic 'zigbee2mqtt/remote', payload '{"action":"arm_day_zones","action_code":"","action_zone":255,"battery":100,"last_seen":"2021-01-31T21:52:51+01:00","linkquality":44}'
Zigbee2MQTT:debug 2021-01-31 21:52:54: Device 'remote' announced itself
Zigbee2MQTT:info  2021-01-31 21:52:54: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"remote","ieee_address":"0x842e14fffef94156"},"type":"device_announce"}'
Zigbee2MQTT:info  2021-01-31 21:52:54: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"remote"},"type":"device_announced"}'
Zigbee2MQTT:debug 2021-01-31 21:52:55: Received Zigbee message from 'remote', type 'commandArm', cluster 'ssIasAce', data '{"armmode":1,"code":"","zoneid":255}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-01-31 21:52:55: MQTT publish: topic 'zigbee2mqtt/remote', payload '{"action":"arm_day_zones","action_code":"","action_zone":255,"battery":100,"last_seen":"2021-01-31T21:52:55+01:00","linkquality":52}'
Zigbee2MQTT:debug 2021-01-31 21:52:58: Received Zigbee message from 'remote', type 'commandArm', cluster 'ssIasAce', data '{"armmode":1,"code":"","zoneid":255}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-01-31 21:52:58: MQTT publish: topic 'zigbee2mqtt/remote', payload '{"action":"arm_day_zones","action_code":"","action_zone":255,"battery":100,"last_seen":"2021-01-31T21:52:58+01:00","linkquality":52}'
Zigbee2MQTT:debug 2021-01-31 21:52:59: Received Zigbee message from 'remote', type 'attributeReport', cluster 'genPowerCfg', data '{"batteryPercentageRemaining":200}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-01-31 21:52:59: MQTT publish: topic 'zigbee2mqtt/remote', payload '{"battery":100,"last_seen":"2021-01-31T21:52:59+01:00","linkquality":52}'
Zigbee2MQTT:debug 2021-01-31 21:53:01: Received Zigbee message from 'remote', type 'commandArm', cluster 'ssIasAce', data '{"armmode":1,"code":"","zoneid":255}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-01-31 21:53:01: MQTT publish: topic 'zigbee2mqtt/remote', payload '{"action":"arm_day_zones","action_code":"","action_zone":255,"battery":100,"last_seen":"2021-01-31T21:53:01+01:00","linkquality":52}'
```

I have two questions:
1. I guess zigbee2mqtt will not (should not) do anything about the repeated messages, and I should handle (debounce it) that in node-red or so? Or do you think there is a protocol issue here, and zigbee2mqtt could/should 'fix' this?
2. You didn't answer about the `e.action` yet. I still think these are correct. What were you suggesting I change about these?

Reply:
1. Very likely the same bug as https://github.com/Koenkk/zigbee2mqtt/issues/5956 
2. If this list matches the published actions  (e.g. `"action":"arm_day_zones"` then it is OK)

Reply:
Fixed your comment about `fz.command_arm`. I assume the bug will be fixed separately then. For now, this seems sufficient for me to add this device (after this has been integrated, I'll also submit a PR for the documentation)

Reply:
Agree, but looks like you changed command_arm for the wrong device. Expected `HM1RC-2-E` but changed for `HS1RC-N` (https://github.com/Koenkk/zigbee-herdsman-converters/pull/2182/commits/5e9c075e75677b8e859e413ea05145f3e5344f50)

Reply:
Oops. Fixed.

Reply:
Updated your pr with a possible fix for the duplicate messages, can you check if it works?

Reply:
Tested. Seems to work perfectly. I now only get a single message whenever I press a button!

Reply:
Great!